workers/event-manager/src/event_types.rs

```diff
@@ -0,0 +1,139 @@
+    transfers::ExternalTransfer,
+    Amount,
+};
+use common::types::wallet::{Order, OrderIdentifier, WalletIdentifier};
+use serde::{Deserialize, Serialize};
+use uuid::Uuid;
+
+// ----------------------------------
+// | Top-level Event / Payload Type |
+// ----------------------------------
+
+/// A system event generated by the relayer
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct RelayerEvent {
+    /// The event ID
+    pub event_id: Uuid,
+    /// The time at which the event occurred
+    pub event_timestamp: SystemTime,
+    /// The event payload itself
+    pub event: RelayerEventPayload,
+}
+
+/// The payload of a system event generated by the relayer.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub enum RelayerEventPayload {
+    /// A wallet creation event
+    WalletCreation(WalletCreationEvent),
+    /// An external transfer event
+    ExternalTransfer(ExternalTransferEvent),
+    /// An order placement event
+    OrderPlacement(OrderPlacementEvent),
+    /// An order update event
+    OrderUpdate(OrderUpdateEvent),
+    /// An order cancellation event
+    OrderCancellation(OrderCancellationEvent),
+    /// A match event
+    Match(MatchEvent),
+    /// An external match event
+    ExternalMatch(ExternalMatchEvent),
+}
+
+// --------------------------
+// | Individual Event Types |
+// --------------------------
+
+/// A wallet creation event
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct WalletCreationEvent {
+    /// The ID of the wallet that was created
+    pub wallet_id: WalletIdentifier,
+}
+
+/// An external transfer event
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct ExternalTransferEvent {
+    /// The ID of the wallet that made the transfer
+    pub wallet_id: WalletIdentifier,
+    /// The transfer that occurred
+    pub transfer: ExternalTransfer,
+}
+
+/// An order placement event
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct OrderPlacementEvent {
+    /// The ID of the wallet that placed the order
+    pub wallet_id: WalletIdentifier,
+    /// The ID of the order that was placed
+    pub order_id: OrderIdentifier,
+    /// The placed order
+    pub order: Order,
+}
+
+/// An order update event
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct OrderUpdateEvent {
+    /// The ID of the wallet that updated the order
+    pub wallet_id: WalletIdentifier,
+    /// The ID of the order that was updated
+    pub order_id: OrderIdentifier,
+    /// The updated order
+    pub order: Order,
+}
+
+/// An order cancellation event
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct OrderCancellationEvent {
+    /// The ID of the wallet that cancelled the order
+    pub wallet_id: WalletIdentifier,
+    /// The ID of the order that was cancelled
+    pub order_id: OrderIdentifier,
+    /// The cancelled order
+    pub order: Order,
+    /// The raw (decimal-unadjusted) remaining amount of the base asset in the
+    /// order
```

Comment:
don't know that we need to call out that it's decimal adjusted considering that's the convention everywhere in the relayer except when fetching prices. it's slightly more confusing this way because it supposes that convention is different